lib/components/narrative/metro/route-block.tsx

```diff
@@ -142,13 +149,15 @@ const RouteBlock = ({
     }
   }
 
+  console.log(leg)
+
   return (
     <>
       {showDivider && previousLegMode && <Divider>•</Divider>}
       <Wrapper className="route-block-wrapper">
         {leg.mode !== previousLegMode && (
           <LegIconWrapper>
-            <LegIcon height={28} leg={leg} />
+            <LegIcon height={28} leg={leg} routeId={leg.routeId || null} />
```

Comment:
Why are we explicitly adding `leg.routeId` when we're already sending the entire `leg`?

Reply:
Yes good point!!